qemu/tests/win_driver_install_by_installer.py

```diff
@@ -0,0 +1,329 @@
+            device_name = "QEMU FWCfg Device"
+            chk_cmd = params["vio_driver_chk_cmd"] % device_name
+            status = session.cmd_status(chk_cmd)
+            if status != 0:
+                test.fail("QEMU FWCfg Device not installed")
+
+        error_context.context("Check if gagent version is correct.",
+                              logging.info)
+        actual_gagent_version = session.cmd_output(gagent_pkg_info_cmd).split()[-2]
+        if actual_gagent_version != expected_gagent_version:
+            test.fail("gagent version is not right, expected is %s but got %s"
+                      % (expected_gagent_version, actual_gagent_version))
+
+        wrong_ver_driver = []
+        not_signed_driver = []
+        for driver_name, device_name in zip(driver_name_list, device_name_list):
+            if driver_name == "qemufwcfg":
+                continue
+            error_context.context("%s Driver Check" % driver_name, logging.info)
+            inf_path = get_driver_inf_path(session, driver_name)
+            expected_ver = session.cmd("type %s | findstr /i /r DriverVer.*=" %
+                                       inf_path, timeout=OPERATION_TIMEOUT)
+            expected_ver = expected_ver.strip().split(",", 1)[-1]
+            if not expected_ver:
+                test.error("Failed to find driver version from inf file")
+            logging.info("Target version is '%s'", expected_ver)
+            ver_list = _pnpdrv_info(session, device_name, ["DriverVersion"])
+            if expected_ver not in ver_list:
+                wrong_ver_driver.append(driver_name)
+            chk_cmd = params["vio_driver_chk_cmd"] % device_name[0:30]
+            chk_output = session.cmd_output(chk_cmd, timeout=chk_timeout)
+            if "FALSE" in chk_output:
+                not_signed_driver.append(driver_name)
+            elif "TRUE" in chk_output:
+                pass
+            else:
+                test.error("Device %s is not found in guest" % device_name)
+        if wrong_ver_driver:
+            test.fail("%s not the expected driver version" % wrong_ver_driver)
+        if not_signed_driver:
+            test.fail("%s not digitally signed!" % not_signed_driver)
+
+    if params.get("run_repaire_test", "no") == "yes":
+        error_context.context("Run virtio-win-guest-tools.exe repaire test",
+                              logging.info)
+        unrepaired_driver = []
+        for driver_name, device_name, device_hwid in zip(driver_name_list,
+                                                         device_name_list, device_hwid_list):
+            if driver_name == "qemufwcfg":
+                if guest_name in ['10', '2016', '2019']:
+                    pass
+                else:
+                    continue
+            error_context.context("Uninstall %s driver" % driver_name, logging.info)
+            uninstall_driver(session, driver_name, device_name, device_hwid)
+            session = vm.reboot(session)
```

Comment:
This is workaround, right? workaround for product bug will casue prodcut bugs missed.
If so, do you confirm dev not plan to fix in the near future? 
If workaround is needed, pls add an explaination in the code.